### PR TITLE
feat(88,105,106,109): Sparkbox Sprint 4

### PR DESCRIPTION
**Pull request by kaseybon**
# READY FOR REVIEW

## Summary
- [STN-88](https://sparkbox.atlassian.net/browse/STN-88) - #456
- [STN-105](https://sparkbox.atlassian.net/browse/STN-105) - #454
- [STN-106](https://sparkbox.atlassian.net/browse/STN-106) - #453
- [STN-109](https://github.com/SU-HSDO/suhumsci/pull/458) - #458

## Need Review By (Date)
asap (1/28)

## Urgency
high

## Steps to Test
- ✅ We've verified that this branch passes the code climate and circle CI tests against the develop branch. (except for the failing php unit test that Mike will add a commit for to this branch)

1. See other PRs if you'd like to test locally for those steps.
2. Verify code looking at the file changes here.
3. Go into humsci_basic folder directory and run `npm i` and `npm run test` to verify tests pass.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

**Review by bryanbraun — APPROVED** [review]
I helped with the individual reviews into this feature branch and I feel comfortable with the changes here. 👍

File: docroot/themes/humsci/humsci_basic/patterns/grid/grid.html.twig [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{%
+  set classes = [
+  'hb-grid',
+  'hb-grid--cols-' ~ columns,
+]
+%}
+
+{% if items|length > 0 %}
+<div{{ create_attribute({'class': classes}) }}>
```
Comment (pookmish) [inline docroot/themes/humsci/humsci_basic/patterns/grid/grid.html.twig]: `attributes` should be an available variable in each twig template. This should just need `attributes.addClass(classes)`

File: docroot/themes/humsci/humsci_basic/patterns/grid/grid.ui_patterns.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+grid:
+  label: "Grid Layout"
+  description: "Grid layout for content area Views and Paragraphs"
+  fields:
```
Comment (pookmish) [inline docroot/themes/humsci/humsci_basic/patterns/grid/grid.ui_patterns.yml:4]: is it possible to provide `preview` for each field? I know some patterns its more difficult to preview than its worth.

**Review by pookmish — CHANGES_REQUESTED** [review]
Just one request in the grid pattern

**Review by pookmish — APPROVED** [review]

File: docroot/themes/humsci/humsci_basic/patterns/grid/grid.ui_patterns.yml (hunk 2 above)
Comment (silverli) [reply docroot/themes/humsci/humsci_basic/patterns/grid/grid.ui_patterns.yml:4]: yep for sure!

Comment (silverli) [reply docroot/themes/humsci/humsci_basic/patterns/grid/grid.ui_patterns.yml:4]: good call

**Pull request merged**